guides/computation-development.md

```diff
@@ -0,0 +1,411 @@
+
+## Basic Overview of COINSTAC System
+
+COINSTAC is an ecosystem for running decentralized algorithms with many clients. It uses a client-server model: while arbitrary calculations can run on clients and servers, the system’s primary use case is [differential privacy](https://en.wikipedia.org/wiki/Differential_privacy), where clients run algorithms with data to ensure their privacy. Clients submit their anonymized results to the server, which amalgamates clients’ data and performs interesting computations, such as averaging, modeling, etc.
+
+COINSTAC’s core is written entirely in JavaScript; writing an algorithm requires writing an entry point in this language.
+ However, the system is robust enough to support algorithms in any language.
+
+| Alpha Warning |
+| ------------- |
+| It’s important to note that *COINSTAC is alpha software*: its rough around the edges. If you have a question or locate a bug please **[open an issue](https://github.com/MRN-Code/coinstac/issues/new)**. |
+
+## Setting Up Your Environment
+
+_It is recommended that you develop in a UNIX-like environment, such as Mac OS X or Linux. While Windows support is planned, it poses several difficulties for algorithm authorship at this time._
+
+Here’s what you’ll need to get up and running with COINSTAC:
+
+1. **Node.js and NPM:** You’ll need the JavaScript runtime to develop for COINSTAC. The easiest way to get it is to [download it via the website](https://nodejs.org/en/download/). Alternatively, you can use [Node Version Manager](https://github.com/creationix/nvm) or [n](https://www.npmjs.com/package/n) to maintain multiple versions of Node.js on the same machine.
+2. **coinstac-simulator:** This will assist in testing your algorithms in the COINSTAC ecosystem. To install, run from a shell:
+
+    ```shell
+    npm install --global coinstac-simulator
+    ```
+
+## How to Create a Computation
+
+Computations in COINSTAC, often referred to as a “decentralized computations” throughout docs, are a set of functions for clients (referred to as “local”) and a central server (referred to as “remote”). These instructions are encapsulated in a JavaScript file that uses a CommonJS export object. Here’s a basic example:
+
+```js
+module.exports = {
+  name: 'my-computation',
+  version: '1.0.0',
+  local: {
+    type: 'function',
+    fn: function(params) {
+
+    },
+  },
+  remote: {
+    type: 'function',
+    fn: function(params) {
+
+    },
+  },
+};
+```
+
+This is the decentralized computation “definition.” The basic properties:
+
+* **`name`** `<String>`: The name of your computation, a JavaScript string of two or more characters. The name uniquely identifies your computation in the COINSTAC ecosystem.
+* **`version`** `<String>`: The release of your computation, a JavaScript string that adheres to [semver](http://semver.org/). This should be incremented in accordance to updates in algorithm code.
+* **`local`** `<Object>` | `<Array>`: Code to be run on clients.
```

Comment:
I know it's probably not a current expectation, however it is not apparent here, or below (at the other comment) that `Array` is an array of computations that form 'steps'